### ISSUE TITLE
TLS error while downloading this step

### ISSUE THREAD

**Issue by ipsy-eric** [bitrise-steplib/steps-hockeyapp-android-deploy#4]
```
WARN[00:12:12] Failed to download step.zip: Get https://bitrise-steplib-collection.s3.amazonaws.com/step-archives/hockeyapp-deploy-android/1.0.0/step.zip: net/http: TLS handshake timeout 
Cloning into '/root/.stepman/step_collections/1470363377/cache/hockeyapp-deploy-android/1.0.0'...
Note: checking out 'c1ab33cbc47825b92ab22e0222d018dd77a80570'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

+------------------------------------------------------------------------------+
| (7) hockeyapp-deploy-android@1.0.0                                           |
+------------------------------------------------------------------------------+
| id: hockeyapp-deploy-android                                                 |
| version: 1.0.0                                                               |
| collection: https://github.com/bitrise-io/bitrise-steplib.git                |
| time: 2016-08-09T00:12:13Z                                                   |
+———————————————————————————————————————+
```

which leads to:

``````
Performing request
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x405cf6]

goroutine 1 [running]:
panic(0x74e480, 0xc82000a150)
    /usr/local/go/src/runtime/panic.go:481 +0x3e6
main.main()
    /tmp/bitrise647193006/step_src/step.go:243 +0x3696
exit status 2
|                                                                              |
+----+--------------------------------------------------------------+----------+
| 🚫  | hockeyapp-deploy-android@1.0.0 (exit code: 1)                | 27 sec   |
+----+--------------------------------------------------------------+----------+
| Issue tracker: ...ithub.com/bitrise-io/steps-hockeyapp-android-deploy/issues |
| Source: https://github.com/bitrise-io/steps-hockeyapp-android-deploy         |
+----+--------------------------------------------------------------+----------+```
``````

**Comment by viktorbenei** [comment]
Thanks for reporting, we'll try to see what/where the issue is.

Is there anything else between the step header (`+———————————————————————————————————————+`) and `Performing request` ?

Note (please comment @ipsy-eric if this is not right): Bitrise CLI version: `1.3.6` I believe, from this log segment it seems the build ran on bitrise.io where the current CLI version is 1.3.6

**Comment by viktorbenei** [comment]
One more question: can we somehow reproduce this, or was this a one time issue @ipsy-eric ?

**Comment by ipsy-eric** [comment]
@viktorbenei for your first question, here's the full logs with sensitive bits redacted:

```
                                         ▼

WARN[00:12:12] Failed to download step.zip: Get https://bitrise-steplib-collection.s3.amazonaws.com/step-archives/hockeyapp-deploy-android/1.0.0/step.zip: net/http: TLS handshake timeout 
Cloning into '/root/.stepman/step_collections/1470363377/cache/hockeyapp-deploy-android/1.0.0'...
Note: checking out 'c1ab33cbc47825b92ab22e0222d018dd77a80570'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

+------------------------------------------------------------------------------+
| (7) hockeyapp-deploy-android@1.0.0                                           |
+------------------------------------------------------------------------------+
| id: hockeyapp-deploy-android                                                 |
| version: 1.0.0                                                               |
| collection: https://github.com/bitrise-io/bitrise-steplib.git                |
| time: 2016-08-09T00:12:13Z                                                   |
+------------------------------------------------------------------------------+
|                                                                              |

Configs:
  apk_path: /bitrise/deploy/app-stagingRelease.apk
  mapping_path: 
  api_token: ***
  app_id: SENSITIVE_BIT
  notes: Deploy with Bitrise HockeyApp Deploy Step.
  notes_type: 0
  notify: 2
  status: 2
  mandatory: 0
  tags: 
  commit_sha: 
  build_server_url: https://www.bitrise.io/build/SENSITIVE_BIT
  repository_url: 

Performing request
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x405cf6]

goroutine 1 [running]:
panic(0x74e480, 0xc82000a150)
    /usr/local/go/src/runtime/panic.go:481 +0x3e6
main.main()
    /tmp/bitrise647193006/step_src/step.go:243 +0x3696
exit status 2
|                                                                              |
+----+--------------------------------------------------------------+----------+
| 🚫  | hockeyapp-deploy-android@1.0.0 (exit code: 1)                | 27 sec   |
+----+--------------------------------------------------------------+----------+
| Issue tracker: ...ithub.com/bitrise-io/steps-hockeyapp-android-deploy/issues |
| Source: https://github.com/bitrise-io/steps-hockeyapp-android-deploy         |
+----+--------------------------------------------------------------+----------+

```

To your second question, immediately re-running the build succeeded. So I wasn't able to reproduce it.

**Comment by ipsy-eric** [comment]
Also FWIW the previous 11 builds, and the 5 subsequent builds, all completed and deployed to Hockey without issues.

**Comment by viktorbenei** [comment]
Thanks, we'll check what could cause it, your logs will help a lot!

@godrei I think the issue is related to the error checking, should be before https://github.com/bitrise-steplib/steps-hockeyapp-android-deploy/blob/c1ab33cbc47825b92ab22e0222d018dd77a80570/step.go#L243 - that's where the exception/panic occurred, based on the log - https://github.com/bitrise-steplib/steps-hockeyapp-android-deploy/blob/c1ab33cbc47825b92ab22e0222d018dd77a80570/step.go#L250 should be before this, or at least `response.Body == nil` should be tested

**Comment by godrei** [comment]
Hi @ipsy-eric ,
the error checking issue is fixed in [v 1.1.0](https://github.com/bitrise-steplib/steps-hockeyapp-android-deploy/releases/tag/1.1.0), it will be available in the steplib in a couple of hours.

**Comment by ipsy-eric** [comment]
Awesome, thanks!
